src/table_properties.rs

```diff
@@ -205,18 +189,12 @@ impl TableProperties {
         self.get_str(DBTableProperty::CompressionName)
     }
 
-    pub fn user_collected_properties(&self) -> HashMap<&[u8], &[u8]> {
-        let mut res = HashMap::new();
-        let mut iter = UserCollectedPropertiesIter::new(self);
-        while iter.valid() {
-            res.insert(iter.key(), iter.value());
-            iter.next();
-        }
-        res
+    pub fn user_collected_properties(&self) -> UserCollectedPropertiesIter {
```

Comment:
I don't get it. `UserCollectedProperties` is a map, how do you access it via index?